[DOC] Add watcher to the threadpool doc

renshuki opened the pull request:
`watcher` is a relatively common threadpool and I think it would be beneficial to add it to the doc.

Based on:
https://github.com/elastic/elasticsearch/blob/68817d7ca29c264b3ea3f766737d81e2ebb4028c/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java#L536-L543

renshuki committed:
Add watcher to the threadpool doc

elasticmachine commented:
Pinging @elastic/es-docs (Team:Docs)

jrodewig commented on docs/reference/modules/threadpool.asciidoc:
```diff
@@ -106,6 +106,10 @@ There are several thread pools, but the important ones include:
     Thread pool type is `fixed` and a default maximum size of
     `min(5, (`<<node.processors, `# of allocated processors`>>`) / 2)`.
 
+`watcher`::
+    For watches execution. Thread pool type is
+    `fixed` with a size of `min(5 * (`<<node.processors, `# of allocated processors`>>`), 50)`, and queue_size of `1000`.
```
```suggestion
`watcher`::
    For <<xpack-alerting,watch executions>>. Thread pool type is `fixed` with a
    size of `min(5 * (`<<node.processors, `# of allocated processors`>>`), 50)`.
    The queue_size is `1000`.
```

renshuki replied:
Awesome, thank you @jrodewig 🙏

jrodewig reviewed (approved):
LGTM. I'm going to incorporate a minor change. I'll wait for an engineer to approve before merging.

jrodewig committed:
Update docs/reference/modules/threadpool.asciidoc

elasticmachine commented:
Pinging @elastic/es-core-features (Team:Core/Features)

joegallo commented:
I'll review this one today, thanks for the heads up @jrodewig.

joegallo committed:
Reword slightly

Just trying to match more precisely the pattern of the previous
entries.

joegallo committed:
Reword all of these slightly

The pattern went awry on all these -- see `management` where it's
"with a keep-alive of `5m` and a default maximum size of". Here we
lost the keep-alive phrase, so the 'and' should instead be a 'with'.

joegallo committed:
Merge branch 'master' into pull/73935

joegallo reviewed (approved):
I added some commits by way of review, so I'm 👍 now. @jrodewig would you mind re-reviewing?

jrodewig reviewed (approved):
LGTM. Thanks @renshuki @joegallo.
